ui/src/map/layers/alignment/location-track-split-alignment-layer.ts

```diff
@@ -90,6 +96,37 @@ function splitToParts(
     });
 }
 
+function findFirstIndexAfterM(points: AlignmentPoint[], m: number): number {
+    const index = points.findIndex((point) => point.m > m);
+    return index === -1 ? points.length : index;
+}
+
+function interpolateWithPrecedingPointOnAlignment(
+    points: AlignmentPoint[],
+    index: number,
+    alignmentM: number,
+): AlignmentPoint[] {
+    const preceding = points[index - 1];
+    const here = points[index];
+    // check vs points being too close to avoid bad precision in interpolation
+    return preceding === undefined || here === undefined || here.m - preceding.m < 0.0001
+        ? []
+        : [linearlyInterpolateAlignmentPoint(preceding, here, alignmentM)];
+}
```

Comment:
Periaatteessa kun tämä palauttaa maksimissaan yhden pisteen, niin tämä voisi palauttaa taulukon sijasta `AlignmentPoint | undefined`. Toki se tarkoittaisi että sitten taas vastaavasti tuolla riveillä 79-83 pitäisi kikkailla että pisteistä saadaan tehtyä kunnollinen taulukko, enkä oikein keksi ainakaan nopeasti ratkaisua mikä olisi kaikkialta kaunis

Reply:
Joo käyttöpaikan ergonomia hieman heikkenee jos täältä palauttaa undefinedia. Paluutyyppiä voisi tässä toki hieman tarkentaa: Esim. `[AlignmentPoint] | []` paluutyyppinä takaisi, että funkkari palauttaa arrayn (tuplen) jossa on tasan 0 tai 1 elementti, mutta tällainen menee hieman hifistelyksi eikä meillä näin tarkkoja olla oltu aiemminkaan

Reply:
Joo, juuri kyseisen kikkailun takia tämä palauttaa taulukon. Saisihan siihen jonkin nullableToList-funktion keksittyä ja sullottua, ja kokeilinkin keksiä ja sulloa, tai vaan käyttää filterNotEmptyä, mutta tällä tavalla sai minusta parhaimman kokonaisuuden kokoon.